website/blog/2022-01-31-avoid-data-silos-in-presto-in-meta.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Avoid Data Silos in Presto in Meta: the journey from Raptor to RaptorX
+author: Rongrong Zhong
+authorURL: https://www.linkedin.com/in/rongrong-zhong-31aa8a8/
+---
+**Alluxio**: Rongrong Zhong
+**Meta**: James Sun, Ke Wang
+
+*Raptor* is a Presto connector ([presto-raptor](https://github.com/prestodb/presto/tree/master/presto-raptor)) that used to power some critical
+interactive query workloads in Meta (previously Facebook). Though referred to in the ICDE 2019 paper, it remains somewhat mysterious to many
+Presto users because  there was no available documentation for this feature. This article will shed some light on the history of Raptor, and why
+Meta eventually replaced  it in favor of a new architecture based on local caching, namely RaptorX.
+
+![Raptor Timeline](/img/blog/2022-01-31-avoid-data-silos-in-presto-in-meta/timeline.png)
+
+## The story of Raptor
+
+Generally speaking, Presto as a query engine does not own storage. Instead, connectors were developed to query different external data sources.
+This framework is very flexible, but in disaggregated compute and storage architectures, it makes low latency guarantees hard to achieve due to the
+network / storage latency. To address this limitation, Raptor is designed as a shared-nothing storage engine for Presto.
+
+### Motivation – an initial use case in the AB testing framework
+
+<!--truncate-->
+
+In Meta, new product features typically  go through AB testing before they are released more broadly. The AB testing framework allows engineers to configure
+experiments by rolling out the feature to a test group, and monitors key metrics against a control group. The framework provides engineers
+with a UI to analyze their experiment’s statistics, which converts the configurations to Presto queries. The query shapes are known and limited.
+Queries typically join multiple large data sets, which include user, device, test, and event attributes, etc. The basic requirements for this use case are:
+1. **Losslessness**: data need to be complete and accurate
```

Comment:
Losslessness sounds awkward and I am not 100% sure why accuracy needs to be called out so here is a random suggestion.
```suggestion
1. **Accurate**: data need to be complete, accurate, and can't be approximate
```

Reply:
I think i used the word from historical documents. Compare to other alternatives like scuba, puma, etc, which can be lossy.